system/scripts/core/custom_commands.cs

```diff
@@ -157,4 +164,73 @@ public CustomCommandResult SCR_CLICK_CHANGEJOB_BUTTON(Character character, int n
 
 	//	return CustomCommandResult.Okay;
 	//}
+
+	[ScriptableFunction("SCR_PUT_CAMPFIRE")]
+	public CustomCommandResult SCR_PUT_CAMPFIRE(Character character, int numArg1, int numArg2, int numArg3)
+	{
+		var area = new Circle(character.Position, 350);
+		var monsters = character.Map.GetAttackableEntitiesIn(character, area);
+
+		foreach (var monster in monsters)
+		{
+			var mob = monster as Mob;
+			// Another camp fire is near by
+			if (mob.Id == 46011)
+			{
+				character.ServerMessage(Localization.Get("There is already a Bonfire placed near by."));
+				return CustomCommandResult.Okay;
+			}
+		}
+
+		this.BonFireSpawn(character);
+
+		return CustomCommandResult.Okay;
+	}
+
+	private async void BonFireSpawn(Character character)
+	{
+		var bonFire = new Mob(46011, MonsterType.NPC);
+		var position = character.Position.GetRelative(character.Direction, 35);
+		bonFire.Position = position;
+		bonFire.Direction = character.Direction;
+
+		character.Map.AddMonster(bonFire);
+
+		Send.ZC_ENTER_MONSTER(bonFire);
+		Send.ZC_FACTION(character.Connection, bonFire, FactionType.Neutral);
+
+		var cancellationTokenSource = new CancellationTokenSource();
+
+		this.AddBuffToNearByPlayers(character, cancellationTokenSource.Token);
+
+		await Task.Delay(TimeSpan.FromMinutes(5));
+
+		cancellationTokenSource.Cancel();
+		Send.ZC_DEAD(bonFire);
+		character.Map.RemoveMonster(bonFire);
+	}
+
+	private async void AddBuffToNearByPlayers(Character character, CancellationToken cancellationToken)
+	{
+		while (!cancellationToken.IsCancellationRequested)
+		{
+			var area = new Circle(character.Position, 350);
+			var characters = character.Map.GeCharactersIn(character, area);
+
+			foreach (var otherCharacter in characters)
+			{
+				if (otherCharacter.IsSitting && !otherCharacter.Buffs.Has(BuffId.campfire_Buff))
```

Comment:
Right, but we could add an event called `PlayerSit` or similar, subscribe the buff handler to it, and then check if there's any campfires nearby when it fires. That way the buff would activate itself in a way, instead of every single campfire spawning a thread to wait for players.